p11/p11.go

```diff
@@ -304,3 +309,113 @@ func (p *p11Token) PrintObjects(label *string) error {
 
 	return nil
 }
+
+func (p *p11Token) GenerateKey(label, keytype string, keysize int) error {
+
+	validRSASize := []int{1024, 2048, 3072, 4096}
+	validAESSize := []int{128, 192, 256}
+	validKeyTypes := []string{"RSA", "AES"}
+
+	if (isValidKeyType(validKeyTypes, keytype)) {
+		switch keytype {
+		case "RSA":
+			if (isValidSize(validRSASize, keysize)) {
+				return p.GenerateRSAKey(label, keysize)
+			} else {
+				return errors.Errorf("Invalid RSA key size: %d", keysize)
+			}
+		case "AES":
+			if (isValidSize(validAESSize, keysize)) {
+				return p.GenerateAESKey(label, keysize)
+			} else {
+				return errors.Errorf("Invalid AES key size: %d", keysize)
+			}
+		}
+	} else {
+		return errors.Errorf("Invalid key type: %s", keytype)
+	}
+
+
+	return nil
+}
+
+func (p *p11Token) GenerateAESKey(label string, keysize int) error {
+
+	privateKeyTemplate := []*pkcs11.Attribute{
+		pkcs11.NewAttribute(pkcs11.CKA_TOKEN, true),
+		pkcs11.NewAttribute(pkcs11.CKA_SIGN, true),
+		pkcs11.NewAttribute(pkcs11.CKA_LABEL, label),
+		pkcs11.NewAttribute(pkcs11.CKA_SENSITIVE, true),
+		pkcs11.NewAttribute(pkcs11.CKA_EXTRACTABLE, true),
+		pkcs11.NewAttribute(pkcs11.CKA_VALUE_LEN, keysize/8),
+	}
+
+	_, err := p.ctx.GenerateKey(p.session,
+		[]*pkcs11.Mechanism{pkcs11.NewMechanism(pkcs11.CKM_AES_KEY_GEN, make([]byte, 16))},
+		privateKeyTemplate)
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("Key \"%s\" generated on token", label)
+
+	return nil
+}
+
+func (p *p11Token) GenerateRSAKey(label string, keysize int) error {
+
+	pubLabel := label + "pub"
+	prvLabel := label + "prv"
+
+	log.Print("Enter GenerateRSAKey" )
```

Comment:
Because we are using a simple logging package (the Go default), everything we log gets printed. There is no option to flag some comments as debug, vs info, etc. As a result, I propose we remove these debugging statements; a normal user of the tool won't need to see them.